### PR TITLE
Us8 choice editor

### PR DESCRIPTION
**Pull request by MorganeL-pro**
Added a new function to edit topics for a user
I redirected the users who have already an active mentoring.
I injected MatchManager for the student with no mentoring yet.

(added fixtures for mentoring (that i already pulled for us22) to check the method)

### CODE REVIEW

**Review by Thomas905 — DISMISSED** [review]

File: src/Controller/ProfileController.php [hunk 1]
```diff
@@ -15,22 +19,28 @@
...
+    public function profile(): Response
     {
         // Fetch UserEmail to get the property IsVerfied
         if ($this->getUser() instanceof UserInterface) {
             $userEmail = $this->getUser()->getUserIdentifier();
-            $user = $userRepository->findOneBy(['email' => $userEmail]);
+            $user = $this->userRepository->findOneBy(['email' => $userEmail]);
             // if it's not verified yet, we redirect the user to home with a flash message
             if ($user === null || ($user !== null && !$user->getIsVerified())) {
                 $this->addFlash(
```
Comment (RomainClair) [inline src/Controller/ProfileController.php]: isVerified is a method from the User class, you should use it directly

File: src/Controller/ProfileController.php [hunk 2]
```diff
@@ -97,6 +107,55 @@ public function edit(
...
+     * @IsGranted("ROLE_USER")
+     */
+    public function editChoice(
+        int $id,
+        Request $request,
+        EntityManagerInterface $entityManager,
+        MatchManager $matchManager
+    ): Response {
+        //fetch the user, the topic and the mentoring
+        $user = $this->checkUser($id);
```
Comment (RomainClair) [inline src/Controller/ProfileController.php]: Again : could be replaced by paramConverter

**Review by RomainClair — CHANGES_REQUESTED** [review]

**Review by Thomas905 — DISMISSED** [review]

**Review by Thomas905 — APPROVED** [review]

File: src/Controller/ProfileController.php [hunk 3]
```diff
@@ -15,24 +18,22 @@
...
-        // Fetch UserEmail to get the property IsVerfied
-        if ($this->getUser() instanceof UserInterface) {
-            $userEmail = $this->getUser()->getUserIdentifier();
-            $user = $userRepository->findOneBy(['email' => $userEmail]);
+        // Fetch User to get the property IsVerfied
+        if ($this->getUser() instanceof User) {
+            $user = $this->getUser();
             // if it's not verified yet, we redirect the user to home with a flash message
-            if ($user === null || ($user !== null && !$user->getIsVerified())) {
+            if ($user == null || ($user !== null && !$user->getIsVerified())) {
```
Comment (RomainClair) [inline src/Controller/ProfileController.php:36]: If $user is an instance of User it's not null

Comment (RomainClair) [reply src/Controller/ProfileController.php:36]: Using lazy operators the right way can probably simplify this test and merge the previous one

**Review by RomainClair — APPROVED** [review]

**Pull request merged**